packages/ibm-products/src/components/CoachmarkOverlayElements/CoachmarkOverlayElements.js

```diff
@@ -0,0 +1,247 @@
+/**
+ * Copyright IBM Corp. 2023, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { Children, useEffect, useRef, useState } from 'react';
+import { Button } from 'carbon-components-react';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import { useCoachmark } from '../Coachmark';
+import { clamp } from 'lodash';
+
+//TODO THIS PATH WILL NEED TO BE UPDATED ONCE IN IBM PRODUCTS
+import { Carousel } from '../Carousel';
+//TODO THIS PATH WILL NEED TO BE UPDATED ONCE IN IBM PRODUCTS
+import { SteppedAnimatedMedia } from '../SteppedAnimatedMedia';
+
+import { getDevtoolsProps } from '../../global/js/utils/devtools';
+import { pkg /*, carbon */ } from '../../settings';
+
+// The block part of our conventional BEM class names (blockClass__E--M).
+const blockClass = `${pkg.prefix}--coachmark-overlay-elements`;
+const componentName = 'CoachmarkOverlayElements';
+
+const defaults = {
+  nextButtonLabel: 'Next',
+  previousButtonLabel: 'Back',
+  closeButtonLabel: 'Got it',
+};
+
+/**
+ * Composable container to allow for the displaying of CoachmarkOverlayElement
+ * components in a carousel fashion.
+ */
+export let CoachmarkOverlayElements = React.forwardRef(
+  (
+    {
+      className,
+      children,
+      media,
+      nextButtonLabel = defaults.nextButtonLabel,
+      previousButtonLabel = defaults.previousButtonLabel,
+      closeButtonLabel = defaults.closeButtonLabel,
+      // Collect any other property values passed in.
+      ...rest
+    },
+    ref
+  ) => {
+    const buttonFocusRef = useRef();
+    const scrollRef = useRef();
+    const [scrollPosition, setScrollPosition] = useState(0);
+    const [currentProgStep, setCurrentProgStep] = useState(0);
+    const coachmark = useCoachmark();
+
+    const numProgSteps = Children.count(children);
+    const progStepFloor = 0;
+    const progStepCeil = numProgSteps - 1;
+
+    useEffect(() => {
+      // On mount, one of the two primary buttons ("next" or "close")
+      // will be rendered and must have focus. (a11y)
+      if (buttonFocusRef.current) {
+        buttonFocusRef.current.focus();
+      }
+    }, []);
+    if (!coachmark) {
```

Comment:
Maybe a console log/error/warn of some kind could be appropriate here if this is a case of incorrect usage?

Reply:
We have a utility for this `pconsole` that could be helpful and is used in some other components as well.